### ISSUE TITLE
Generic.Commenting.DocComment.MissingShort is too broad

### ISSUE THREAD

**Issue by Rarst** [WordPress-Coding-Standards/WordPress-Coding-Standards#403]
From a stab at making existing plugin adhere to new WordPress-Docs ruleset, the sniff enforcing short descriptions in doc blocks is overly broad.

That would make sense to me if limited to function/property descriptions.

However it catches multiple _other_ kinds of blocks:

 - file headers (where description can be redundant)
 - `/** @var ...` blocks of variables and properties documentation
 - `/** @todo ...` and other arbitrary blocks

**Comment by GaryJones** [comment]
Summaries in File headers are required per WPCS: https://make.wordpress.org/core/handbook/inline-documentation-standards/php-documentation-standards/#6-file-headers

Summaries in Class properties are required per WPCS: https://make.wordpress.org/core/handbook/inline-documentation-standards/php-documentation-standards/#2-class-properties - nothing about documenting variables though.

A `@todo` in it's own DocBlock serves little purpose (I know it's the only way for phpDocumentor to see it) - either leave it as `//` standard comment, or create a GH Issue for it so it's not lost in the code.

(I agree that the CS in the handbook may be too stringent, but WPCS is only codifying what's there.)

**Comment by Rarst** [comment]
Sniffs should not go above the letter of the standard. Enforcing short descriptions on blocks that aren't intended to have them is sniff issue, not something _code_ should bend to accommodate.

**Comment by GaryJones** [comment]
Agreed, but equally, code should be used correctly in the first place. I _could_ add a `/**` DocBlock for every single inline comment, but that is neither standard, not reasonably expected. PSR-5 (for want of better evidence) articulates that `@todo` has an _associated Structural Element_` That means `@todo` should only be added to a standard DocBlock, not added as what would otherwise be an inline comment, mid-function.

**Comment by Rarst** [comment]
`/** @var ...` is perfectly correct usage (explicitly used in example in PSR-5 like that). We can argue _some_ of this away, but in the end I don't think the sniff works in reasonable and productive way at the moment.

**Comment by GaryJones** [comment]
There doesn't appear to be a way to exclude this check for different types of structural elements. The whole sniff gives lots of advantages though, and not everyone uses `/** @var ...`, so, for the time being, I suggest you just exclude that particular check in your projects.

**Comment by GaryJones** [comment]
Relevant: https://github.com/squizlabs/PHP_CodeSniffer/issues/258

**Comment by bkdotcom** [comment]
> `/** @var ...` is perfectly correct usage

as is
```php
	/**
	 * @var $myVar ...
	 */
```

just to be clear that this isn't single-line syntax specific

**Comment by larssn** [comment]
I use `/** @var ...` extensively, and writing `// phpcs:ignore` kinda defeats the purpose.

Can we get some momentum going on this issue as it's very old at this point.

**Comment by jrfnl** [comment]
@larssn This is a sniff which comes from PHPCS itself, not one maintained here. You're welcome to send a PR to PHPCS itself to see about fixing it.
